woodwork/type_sys/type_system.py

```diff
@@ -154,41 +166,57 @@ def add_type(self, logical_type, inference_function=None, parent=None):
             parent=parent,
         )
 
-        registered_ltype_names = [ltype.__name__ for ltype in self.registered_types]
-        if logical_type.__name__ in registered_ltype_names:
+        registered_ltype_names = {ltype.__name__ for ltype in self.registered_types}
+        ltype_is_registered = logical_type.__name__ in registered_ltype_names
+        if ltype_is_registered and treatment is None:
```

Comment:
should we explicitly check that the types passed in are either `ignore`, `replace`, or `none`? In this case, a user could be passing in some typo, and the default behavior seems to treat it the same as `ignore`, which would be an issue.